### DISCUSSION TITLE
E5113 error during install

### DISCUSSION THREAD

**Discussion by spinkyspider** [lumen-oss/rocks.nvim#367] [Q&A] [answered]
Problem

Alpine Linux v3.21
neovim v.0.10.0
Installing using the Bootstrap script

Downloading luarocks...
Configuring luarocks...
Error detected while processing /home/churk/.config/nvim/init.lua:
E5113: Error while calling lua chunk: /home/churk/.config/nvim/init.lua:53: Vim:Configuring luarocks failed.
stack traceback:
[C]: in function 'source'
/home/churk/.config/nvim/init.lua:53: in main chunk

[init.lua.zip](https://github.com/user-attachments/files/15725927/init.lua.zip)

### Steps to reproduce

put the bootstrap script in ~/.config/nvim/init.lua
then run 'nvim'

### Expected behavior

I had hoped luarocks would have installed and worked. :-/

NVIM v0.10.0
Build type: MinSizeRel
LuaJIT 2.1.1710398010

### Neovim version (nvim -v)

NVIM v0.10.0 Build type: MinSizeRel

### Vim (not Nvim) behaves the same?

n/a

### Operating system/version

Alpine Linux v3.21

### Terminal name/version

kitty 0.31.0

### $TERM environment variable

~ $ echo $TERM --> xterm-kitty

### Installation

Alpine Linux Package manager

**Comment by mrcjkb** [comment]
Hey :wave: 

> Configuring luarocks failed.

tells me that the following command failed during the luarocks installation:

```bash
sh configure --prefix=<rocks_path> --lua-version=5.1 --force-config
```

where `<rocks_path>` is set in your init.lua.
Perhaps you're missing some dependencies?
:thinking: it didn't output the command's `stderr` or `stdout`, which is strange.

**Comment by spinkyspider** [comment]
I don't reall know what the rocks path is.  I found a webpage that says 'echo $VIMRUNTIME' and, inside neovim, it gives me the following:
```
:!echo $VIMRUNTIME                                                                                               
/usr/share/nvim/runtime  
```
So, I did the following:
```
~ $ sh configure --prefix=/usr/share/nvim/runtime --lua-version=5.1 --force-config
sh: can't open 'configure': No such file or directory
~ $ which configure
~ $ 
```

So, I guess I'm missing something.  Should the 'configure' be run in a particular directory?

**Reply by mrcjkb** [reply]
The bootstrap script contains the following:

```lua
    -- Specifies where to install/use rocks.nvim
    local install_location = vim.fs.joinpath(vim.fn.stdpath("data"), "rocks")

    -- Set up configuration options related to rocks.nvim (recommended to leave as default)
    local rocks_config = {
        rocks_path = vim.fs.normalize(install_location),
        luarocks_binary = vim.fs.joinpath(install_location, "bin", "luarocks"),
    }
```

Until recently, rhe bootstrap script cloned luarocks into `rocks_path` and ran `sh configure` in `<rocks_path>/luarocks`.
This has changed recently. Now, it clones luarocks into a temporary directory. 

To troubleshoot this, you can try [cloning luarocks](https://github.com/luarocks/luarocks) into a temporary directory and running the command from there.

**Reply by pacidic** [reply, marked as answer]
Had this issue, turns out it was due to missing lua5.1 headers. On ubuntu 22.04, running `sudo apt install liblua5.1-0-dev` fixed it.
